File: TransCoder_tokenized_test_set_functions/cpp/FIND_THE_NUMBER_OCCURRING_ODD_NUMBER_OF_TIMES_1.cpp
int getOddOccurrence ( int arr [ ], int size ) {
  unordered_map < int, int > hash;
  for ( int i = 0;
  i < size;
  i ++ ) {
    hash [ arr [ i ] ] ++;
  }
  for ( auto i : hash ) {
    if ( i . second % 2 != 0 ) {
      return i . first;
    }
  }
  return - 1;
}

File: TransCoder_tokenized_test_set_functions/java/FIND_THE_NUMBER_OCCURRING_ODD_NUMBER_OF_TIMES_1.java
static int getOddOccurrence ( int arr [ ], int n ) {
  HashMap < Integer, Integer > hmap = new HashMap < > ( );
  for ( int i = 0;
  i < n;
  i ++ ) {
    if ( hmap . containsKey ( arr [ i ] ) ) {
      int val = hmap . get ( arr [ i ] );
      hmap . put ( arr [ i ], val + 1 );
    }
    else hmap . put ( arr [ i ], 1 );
  }
  for ( Integer a : hmap . keySet ( ) ) {
    if ( hmap . get ( a ) % 2 != 0 ) return a;
  }
  return - 1;
}